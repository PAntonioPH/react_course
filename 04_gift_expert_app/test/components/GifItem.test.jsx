import {render, screen} from "@testing-library/react";
import {GifItem} from "../../src/components/index.js";

describe("Pruebas de GifItem", () => {
  const testTitle = "Naruto"
  const testUrl = "https://media3.giphy.com/media/naruto"

  test("De de hacer match con el snapshot", () => {
    const {container} = render(<GifItem title={testTitle} url={testUrl}/>)

    expect(container).toMatchSnapshot()
  })

  test("Debe de mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GifItem title={testTitle} url={testUrl}/>)
    // screen.debug()
    // expect(screen.getByRole("img").alt).toBe(testTitle)

    const {src, alt} = screen.getByRole("img")

    expect(src).toBe(testUrl)
    expect(alt).toBe(testTitle)
  })

  test("Debe de mostrar el titulo en el componente", () => {
    render(<GifItem title={testTitle} url={testUrl}/>)

    const {innerHTML} = screen.getByText(testTitle)
    expect(screen.getByText(testTitle)).toBeTruthy()
    expect(innerHTML).toBe(testTitle)
  })
})