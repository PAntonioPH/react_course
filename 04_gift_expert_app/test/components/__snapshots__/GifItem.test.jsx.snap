// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifItem De de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      Naruto
    </p>
    <img
      alt="Naruto"
      src="https://media3.giphy.com/media/naruto"
    />
  </div>
</div>
`;
